File: NetCoreWithAngular/Models/Book.cs
using CoreLibrary;
using System.ComponentModel.DataAnnotations;

namespace NetCoreWithAngular.Models
{
    public enum Genre
    {
        [Display(Name ="Genre.Fantasy")]
        Fantasy = 1,
        [Display(Name = "Genre.Horror")]
        Horror = 2,
        [Display(Name = "Genre.Drama")]
        Drama = 3,
        [Display(Name = "Genre.Sci-Fi")]
        SciFi = 4
    }
    public class Book : IEntity<int>
    {
        [Required]
        public int Id { get; set; }

        public string Title { get; set; }

        public string Author { get; set; }

        public int? PageCount { get; set; }

        public Genre? Genre { get; set; }
    }
}

File: NetCoreWithAngular/ClientApp/app/Admin/Models/Book/book.ts
import { IEntity } from '../../../../Core/Models/IEntity'
import { SelectListItem } from '../../../../Core/Models/SelectListItem';

enum Genre {
    Fantasy = 1,
    Horror = 2,
    Drama = 3,
    SciFi = 4
}

export class Book implements IEntity<number> {
    id?: number;
    title: string;
    author: string;
    pageCount: number;
    genre?: Genre;
    genreList: SelectListItem[];
}